api/test/json-api.e2e-spec.ts

```diff
@@ -64,4 +107,57 @@ describe('JSON API Specs (e2e)', () => {
       }
```

Comment:
Not really related to this PR but I believe this isn't a reliable test. CI runs in a particular environment and any `if`s (especially there, that it may omit them at all) may cause false-positive test.
If a CI runs with `prod` for example as NODE_ENV, it will skip any assertions.

In particular, having those `if`s imply that we never check the second scenario

Reply:
Have to say that I am agree at this point. And I am personally not a fan of testing what an app does depending on enviroment, I think we should focus quality to production, and I'd rather check this logics in unit testing. But we talked about doing this, and having separated test for different enviroments required to stop, change env and re-run the test-app. 
Any better ways to achieve this?